### PR TITLE
[ServiceFabricManagedClusters] Polish the API 

### PR DESCRIPTION
**Pull request by Yao725**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/servicefabricmanagedclusters/Azure.ResourceManager.ServiceFabricManagedClusters/api/Azure.ResourceManager.ServiceFabricManagedClusters.netstandard2.0.cs [hunk 1]
```diff
@@ -1,474 +1,365 @@
...
+        public static Azure.AsyncPageable<Azure.ResourceManager.ServiceFabricManagedClusters.Models.ServiceFabricManagedClusterVersion> GetManagedClusterVersionsByEnvironmentAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.Core.AzureLocation location, Azure.ResourceManager.ServiceFabricManagedClusters.Models.ManagedClusterVersionEnvironment environment, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Response<Azure.ResourceManager.ServiceFabricManagedClusters.Models.ServiceFabricManagedClustersVmSize> GetManagedUnsupportedVmSize(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.Core.AzureLocation location, string vmSize, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (fengzhou-msft) [inline sdk/servicefabricmanagedclusters/Azure.ResourceManager.ServiceFabricManagedClusters/api/Azure.ResourceManager.ServiceFabricManagedClusters.netstandard2.0.cs:343]: Shall we rename the response model to `ServiceFabricManagedUnsupportedVmSize`?

File: sdk/servicefabricmanagedclusters/Azure.ResourceManager.ServiceFabricManagedClusters/api/Azure.ResourceManager.ServiceFabricManagedClusters.netstandard2.0.cs [hunk 2]
```diff
@@ -1,474 +1,365 @@
 namespace Azure.ResourceManager.ServiceFabricManagedClusters
 {
-    public partial class ApplicationResource : Azure.ResourceManager.ArmResource
+    public partial class ServiceFabricManagedClusterApplicationCollection : Azure.ResourceManager.ArmCollection, System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.ServiceFabricManagedClusters.ServiceFabricManagedClusterApplicationResource>, System.Collections.Generic.IEnumerable<Azure.ResourceManager.ServiceFabricManagedClusters.ServiceFabricManagedClusterApplicationResource>, System.Collections.IEnumerable
```
[the remark this answers is not in the document]
Comment (Yao725) [reply sdk/servicefabricmanagedclusters/Azure.ResourceManager.ServiceFabricManagedClusters/api/Azure.ResourceManager.ServiceFabricManagedClusters.netstandard2.0.cs]: Fixed

File: sdk/servicefabricmanagedclusters/Azure.ResourceManager.ServiceFabricManagedClusters/api/Azure.ResourceManager.ServiceFabricManagedClusters.netstandard2.0.cs (hunk 1 above)
Comment (Yao725) [reply sdk/servicefabricmanagedclusters/Azure.ResourceManager.ServiceFabricManagedClusters/api/Azure.ResourceManager.ServiceFabricManagedClusters.netstandard2.0.cs]: Renamed.

**Pull request merged**